src/D2L.Bmx/Program.cs

```diff
@@ -147,23 +138,37 @@
 		consolePrompter,
 		config
 	);
-	try {
-		await handler.HandleAsync(
-			org: context.ParseResult.GetValueForOption( orgOption ),
-			user: context.ParseResult.GetValueForOption( userOption ),
-			account: context.ParseResult.GetValueForOption( accountOption ),
-			role: context.ParseResult.GetValueForOption( roleOption ),
-			duration: context.ParseResult.GetValueForOption( durationOption ),
-			nonInteractive: context.ParseResult.GetValueForOption( nonInteractiveOption ),
-			output: context.ParseResult.GetValueForOption( writeOutputOption ),
-			profile: context.ParseResult.GetValueForOption( profileOption )
-		);
-	} catch( BmxException e ) {
-		Console.Error.WriteLine( e.Message );
-		context.ExitCode = 1;
-	}
-} );
+	return handler.HandleAsync(
+		org: context.ParseResult.GetValueForOption( orgOption ),
+		user: context.ParseResult.GetValueForOption( userOption ),
+		account: context.ParseResult.GetValueForOption( accountOption ),
+		role: context.ParseResult.GetValueForOption( roleOption ),
+		duration: context.ParseResult.GetValueForOption( durationOption ),
+		nonInteractive: context.ParseResult.GetValueForOption( nonInteractiveOption ),
+		output: context.ParseResult.GetValueForOption( writeOutputOption ),
+		profile: context.ParseResult.GetValueForOption( profileOption )
+	);
+} ) );
 
 rootCommand.Add( writeCommand );
 
 return await rootCommand.InvokeAsync( args );
+
+static async Task RunWithErrorHandlingAsync( InvocationContext context, Func<Task> handle ) {
+	try {
+		await handle();
+	} catch( Exception e ) {
+		Console.ResetColor();
+		Console.ForegroundColor = ConsoleColor.Red;
```

Comment:
not sure why the color needs to be reset before setting it to what we want, but it's what System.CommandLine does internally
https://github.com/dotnet/command-line-api/blob/2.0.0-beta4.22272.1/src/System.CommandLine/Invocation/ParseErrorResult.cs#L21-L33